docs/src/formula.md

```diff
@@ -14,167 +14,184 @@ fields with possibly heterogeneous types.  One of the primary goals of
+with the formula operators `+`, `&`, and `~`:
 
-```jldoctest
-julia> lhs = :y
-:y
+```@repl 1
+Term(:y) ~ ConstantTerm(1) + Term(:a) + Term(:b) + Term(:a) & Term(:b)
 ```
 
-and you want to create a formula whose left-hand side is the _value_ of `lhs`,
-as in
+The [`term`](@ref) function constructs a term of the appropriate type from
+symbols and numbers, which makes it easy to work with collections of mixed type:
 
-```jldoctest
-julia> @formula(y ~ 1 + x)
-Formula: y ~ 1 + x
+```@repl 1
+ts = term(1, :a, :b)
 ```
 
-Simply using the Julia interpolation syntax `@formula($lhs ~ 1 + x)` won't work,
-because `@formula` runs _at compile time_, before anything about the value of
-`lhs` is known.  Instead, you need to construct and evaluate the _correct call_
-to `@formula`.  The most concise way to do this is with `@eval`:
+These can then be combine with standard reduction techniques:
 
-```jldoctest
-julia> @eval @formula($lhs ~ 1 + x)
-Formula: y ~ 1 + x
+```@repl 1
+f1 = term(:y) ~ foldl(+, ts)
+f2 = term(:y) ~ sum(ts)
+f1 == f2 == @formula(y ~ 1 + a + b)
 ```
 
-The `@eval` macro does two very different things in a single, convenient step:
+## Fitting a model from a formula
 
-1. Generate a _quoted expression_ using `$`-interpolation to insert the run-time
-   value of `lhs` into the call to the `@formula` macro.
-2. Evaluate this expression using `eval`.
+The main use of `@formula` is to streamline specifying and fitting statistical
+models based on tabular data.  From the user's perspective, this is done by
+`fit` methods that take a `FormulaTerm` and a table instead of numeric
+matrices.
 
-An equivalent but slightly more verbose way of doing the same thing is:
+As an example, we'll simulate some data from a linear regression model with an
+interaction term, a continuous predictor, a categorical predictor, and the
+interaction of the two, and then fit a `GLM.LinearModel` to recover the
+simulated coefficients.
 
-```jldoctest
-julia> formula_ex = :(@formula($lhs ~ 1 + x))
-:(@formula y ~ 1 + x)
+```julia-repl
+julia> using GLM, DataFrames, StatsModels
 
-julia> eval(formula_ex)
-Formula: y ~ 1 + x
-```
+julia> data = DataFrame(a = rand(100), b = repeat(["d", "e", "f", "g"], 25));
 
-### Technical details
+# simulate response:
+julia> X = modelmatrix(@formula(y ~ 1 + a*b).rhs, data);
 
-You may be wondering why formulas in Julia require a macro, while in R they
-appear "bare."  R supports nonstandard evaluation, allowing the formula to
-remain an unevaluated object while its terms are parsed out. Julia uses a much
-more standard evaluation mechanism, making this impossible using normal
-expressions. However, unlike R, Julia provides macros to explicitly indicate
-when code itself will be manipulated before it's evaluated. By constructing a
-formula using a macro, we're able to provide convenient, R-like syntax and
-semantics.
+julia> β_true = [1., 2, 3, 4, 5, 6, 7, 8];
```

Comment:
```suggestion
julia> β_true = 1:8;
```

Reply:
I never know where I can safely use a range in place of a vector so I just superstitiously always use a vector (as you've seen)

Reply:
These days ranges should work in most situations AFAIK. The main difference used to be equality, but we've fixed that.